addons/web/static/src/js/views/list/list_confirm_dialog.js

```diff
@@ -68,25 +68,53 @@ const ListConfirmDialog = Dialog.extend(WidgetAdapterMixin, {
                 options: widgetOptions,
             });
         }
+
+        this.relatedFieldWidget = undefined;
+        if (this.fieldWidget.attrs.widget === 'daterange' && changes.relatedFieldName) {
```

Comment:
Should we really bound the logic to the daterange widget or could we potentially have other widgets that would update several fields values? To see with the JS team if they prefer this or adding a property on the widgets that would inform about that behavior.

Reply:
Yeah, you are right ! 
I can refactor it and avoid duplicate like this: 
```js
        const Widget = record.fieldsInfo.list[changes.fieldName].Widget;
        const widgetOptions = {
            mode: 'readonly',
            viewType: 'list',
            noOpen: true,
        };
        this.isLegacyWidget = !utils.isComponent(Widget);

        const generateFieldWidget = (Widget, fieldName) => {
            if (this.isLegacyWidget) {
                return new Widget(this, fieldName, record, widgetOptions);
            } else {
                return new FieldWrapper(this, Widget, {
                    fieldName: fieldName,
                    record,
                    options: widgetOptions,
                });
            }
        };

        this.fieldWidget = generateFieldWidget(Widget, changes.fieldName);

        this.relatedFieldWidget = null;
        if (this.fieldWidget.attrs.widget === 'daterange' && changes.relatedFieldName) {
            const fieldName = changes.relatedFieldName;
            const fieldLabel = record.fields[fieldName].string;
            this.$content.find('table.o_modal_changes tbody').append(qweb.render('ListView.confirmModal.relatedField', { fieldLabel }));
            const RelatedWidget = record.fieldsInfo.list[fieldName].Widget;

            // the related widget is the same (= DateRange widget) then we don't need to recheck it
            this.relatedFieldWidget = generateFieldWidget(RelatedWidget, fieldName);
        }
```

Reply:
this is an interesting question,  I think some other fieldWidget have a corresponding field that should update with those kind of interactions. In those cases, there is an options in the fieldInfo.
Admittedly, the API of those Widget taking two fields rather than one is not ideal

Reply:
I only know the daterange which needs another field to correctly work. To add the second field, we need to define the name of the second field in the options attribute in the first field with the key called "related_field". 

Indeed, the daterange is the only one and I don't think we could have others widgets need more than one field. 🙂